src/classes/VOL_CTRL_VolunteersCampaignWizard.cls

```diff
@@ -117,267 +119,263 @@ public with sharing class VOL_CTRL_VolunteersCampaignWizard {
+        list<Volunteer_Shift__c> listShifts = new list<Volunteer_Shift__c>();
+        list<Volunteer_Hours__c> listHours = new list<Volunteer_Hours__c>();
+        list<Volunteer_Job__c> listJobsExisting = new list<Volunteer_Job__c>();
+        list<Volunteer_Shift__c> listShiftsExisting = new list<Volunteer_Shift__c>();
+        list<Volunteer_Hours__c> listHoursExisting = new list<Volunteer_Hours__c>();
+
+        // initialize the list of all fields on Job.
+        Map<String, Schema.SObjectField> mapS = Schema.SObjectType.Volunteer_Job__c.fields.getMap();
+        list<string> listStrFieldsJob = new list<string>();
+        listStrFieldsJob.addAll(mapS.keySet());
+
+        // Ensure the user has access to the object and fields before querying
+        UTIL_Describe.checkReadAccess(UTIL_Describe.StrTokenNSPrefix('Volunteer_Job__c'),
+            new Set<String>(listStrFieldsJob));
+
+        // find the existing jobs
+        // we need to use dynamic soql to pick up all custom fields we don't know about
+        string strSoql = 'select ';
+        string strComma = '';
+        for (string strF : listStrFieldsJob) {
+            strSoql += strComma + strF;
+            strComma = ', ';
+        }
+        strSoql += ' from Volunteer_Job__c where Campaign__c = :campaignIdClone order by CreatedDate';
+        listJobsExisting = Database.Query(strSoql);
+
+        // initialize the list of all fields on Shift.
+        mapS = Schema.SObjectType.Volunteer_Shift__c.fields.getMap();
+        list<string> listStrFieldsShift = new list<string>();
+        listStrFieldsShift.addAll(mapS.keySet());
+
+        // Ensure the user has access to the object and fields before querying
+        UTIL_Describe.checkReadAccess(UTIL_Describe.StrTokenNSPrefix('Volunteer_Shift__c'),
+            new Set<String>(listStrFieldsShift));
+
+        // find the existing shifts for those jobs
+        // we need to use dynamic soql to pick up all custom fields we don't know about
+        strSoql = 'select ';
+        strComma = '';
+        for (string strF : listStrFieldsShift) {
+            strSoql += strComma + strF;
+            strComma = ', ';
+        }
+        strSoql += ' from Volunteer_Shift__c where Volunteer_Job__c in :listJobsExisting order by Start_Date_Time__c';
+        listShiftsExisting = Database.Query(strSoql);
+
+        // clone the jobs
+        for (Volunteer_Job__c job : listJobsExisting) {
+            Volunteer_Job__c jobNew = new Volunteer_Job__c();
+
+            //  now copy over all the non-null fields
             for (string strF : listStrFieldsJob) {
-                strSoql += strComma + strF;
-                strComma = ', ';
+                if (strF != 'Id' && job.get(strF) != null) {
+                    try {
+                        jobNew.put(strF, job.get(strF));
+                    } catch (exception ex) {
+                        // catch errors from readonly fields, and continue.
```

Comment:
Is this try/catch block needed? Avoid swallowing exceptions where possible.

Reply:
yes, as the comment says to deal with readonly fields.  that's because the list of fields contains all fields on the object.